src/d4bl/app/api.py

```
@@ -41,69 +45,91 @@
-    return _query_engine
+    """Return a cached QueryEngine singleton for NL query processing."""
+    return QueryEngine()
 
-app = FastAPI(title="D4BL AI Agent API", version="1.0.0")
 
-# Initialize database on startup
-@app.on_event("startup")
-async def startup_event():
-    """Initialize database on application startup"""
+@asynccontextmanager
+async def lifespan(app: FastAPI):
```

Ruff (0.15.2)

[warning] Unused function argument: `app`

(ARG001)